### PR TITLE
K8SPSMDB-489: Don't update status of ready backups

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPSMDB-489](https://jira.percona.com/browse/K8SPSMDB-489) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by delgod — APPROVED** [review]

File: pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go [hunk 1]
```diff
@@ -106,6 +106,9 @@ func (r *ReconcilePerconaServerMongoDBBackup) Reconcile(request reconcile.Reques
 	status := cr.Status
 
 	defer func() {
+		if cr.Status.State == psmdbv1.BackupStateReady {
```
Comment (delgod) [inline pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go:109]: why it is inside defer?
I think we should not reconcile any CR which already has an error or ready state (like place the same code after 104 line

**Review by delgod — CHANGES_REQUESTED** [review]

File: pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go (hunk 1 above)
Comment (egegunes) [reply pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go]: https://github.com/percona/percona-server-mongodb-operator/pull/697/commits/a1108587aca5267e1c585320b73bdfa1c2d65799

**Review by delgod — APPROVED** [review]

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| init-deploy | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| arbiter | passed |
| security-context | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| self-healing-chaos | passed |
| liveness | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| operator-self-healing | passed |
| pitr | passed |
| operator-self-healing-chaos | passed |
| pitr-sharded | failed |
| smart-update | passed |
| version-service | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/697/commits/98f992855ac612f9e793ef9f9b118e4e7b7e2327
image: `perconalab/percona-server-mongodb-operator:PR-697-98f99285`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| limits | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| self-healing-chaos | passed |
| scheduled-backup | passed |
| service-per-pod | passed |
| rs-shard-migration | passed |
| liveness | passed |
| operator-self-healing | passed |
| demand-backup-sharded | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/697/commits/a1108587aca5267e1c585320b73bdfa1c2d65799
image: `perconalab/percona-server-mongodb-operator:PR-697-a1108587`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| limits | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| rs-shard-migration | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/697/commits/1bd3aa7cd494a7f2b980c4919bfeb44a28f9477f
image: `perconalab/percona-server-mongodb-operator:PR-697-1bd3aa7c`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| demand-backup | passed |
| limits | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| scheduled-backup | passed |
| security-context | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| self-healing-chaos | passed |
| service-per-pod | passed |
| operator-self-healing | passed |
| liveness | passed |
| smart-update | passed |
| operator-self-healing-chaos | passed |
| version-service | passed |
| demand-backup-sharded | passed |
| users | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| data-sharded | passed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/697/commits/a65b85af02baa000d72ee347a77c86e8b199a4ea
image: `perconalab/percona-server-mongodb-operator:PR-697-a65b85af`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| monitoring-2-0 | passed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| init-deploy | passed |
| users | passed |
| limits | passed |
| data-sharded | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| pitr | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/697/commits/a65b85af02baa000d72ee347a77c86e8b199a4ea
image: `perconalab/percona-server-mongodb-operator:PR-697-a65b85af`

**Pull request merged**